timemachine/cpp/src/wrap_kernels.cpp

```diff
@@ -381,6 +381,173 @@ void declare_potential(py::module &m) {
             py::arg("params"),
             py::arg("box"),
             py::arg("lam"))
+        .def(
+            "execute_selective_batch",
+            [](timemachine::Potential &pot,
+               const py::array_t<double, py::array::c_style> &coords,
+               const py::array_t<double, py::array::c_style> &params,
+               const py::array_t<double, py::array::c_style> &boxes,
+               const py::array_t<double, py::array::c_style> &lambdas,
+               const bool compute_du_dx,
+               const bool compute_du_dp,
+               const bool compute_du_dl,
+               const bool compute_u) -> py::tuple {
+                if (coords.ndim() != 3 && boxes.ndim() != 3) {
+                    throw std::runtime_error("coords and boxes must have 3 dimensions");
+                }
+                if (coords.shape()[0] != boxes.shape()[0]) {
+                    throw std::runtime_error("number of batches of coords and boxes don't match");
+                }
+                if (params.ndim() < 2) {
+                    throw std::runtime_error("parameters must have at least 2 dimensions");
+                }
+                if (lambdas.size() < 1) {
+                    throw std::runtime_error("lambdas must have at least 1 value");
+                }
+                const long unsigned int coord_batches = coords.shape()[0];
+                const long unsigned int N = coords.shape()[1];
+                const long unsigned int D = coords.shape()[2];
```

Comment:
Would it be useful to assert D is 3 or 4? (In case user-provided array has erroneous shape.)

Reply:
I wanted to leave that sort of logic to the potential, as we have checks at that level. Wanted to avoid too much repetition of checks and leave the most stringent checks to the derived potentials.